CBOR/PeterO/Cbor/CBORObject.cs

```diff
@@ -568,7 +569,8 @@ private object ThisItem
     /// or map. If this is a CBOR map, returns <c>null</c> (not
     /// <c>CBORObject.Null</c> ) if an item with the given key doesn't
     /// exist.</returns>
-    public CBORObject GetOrDefault(object key, CBORObject defaultValue) {
+    [RequiresUnreferencedCode("Do not use in AOT or reflection-free contexts.")]
+    public CBORObject GetOrDefault(object key, CBORObject defaultValue) { // Suspect method. This is always used with a string key in the main library. But if the key is not a number it will return defaultValue.
```

Comment:
@peteroupc can you look at this comment?

Reply:
Since the next release is expected to be major, it's appropriate to change this method to accept only strings, `CBORObject`, and a limited selection of integer types in this method.

Reply:
Understood. But did I identify a logical issue in the current code?

> // Suspect method. This is always used with a string key in the main library. But if the key is not a number it will return defaultValue.

Issue is that if the key is a string (which it always is inside the library) it will go into this branch

```
          if (!cborkey.IsNumber || !cborkey.AsNumber().CanFitInInt32()) {
            return defaultValue;
          }
```
and always return defaultValue.

Reply:
CBOR objects can be arrays or maps, among other things.  Arrays take only integer keys; for maps, the keys can be strings or any other CBOR object.  If any CBOR object for which `GetOrDefault` is called is an array, and the key is not a number, then that is a logical error; not so if the CBOR object is a map.

Reply:
Oops apologies